lib/ansible/modules/network/checkpoint/cp_gaia_physical_interface.py

```diff
@@ -0,0 +1,158 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+author: Yuval Feiger (@chkp-yuvalfe)
+description:
+- Set Physical interface
+module: cp_gaia_physical_interface
+options:
+  auto_negotiation:
+    description: Activating Auto-Negotiation will skip the speed and duplex configuration
+    required: false
+    type: bool
+  comments:
+    description: interface Comments.
+    required: false
+    type: str
+  duplex:
+    description: Duplex for the interface. Duplex is not relevant when 'auto-negotiation' is enabled
+    required: false
+    type: str
+  enabled:
+    description: interface State.
+    required: false
+    type: bool
+  ipv4_address:
+    description: Interface IPv4 address.
+    required: false
+    type: str
+  ipv4_mask_length:
+    description: Interface IPv4 address mask length.
+    required: false
+    type: str
+  ipv6_address:
+    description: Interface IPv6 address.
+    required: false
+    type: str
+  ipv6_autoconfig:
+    description: Configure IPv6 auto-configuration.
+    required: false
+    type: bool
+  ipv6_mask_length:
+    description: Interface IPv6 address mask length.
+    required: false
+    type: str
+  mac_addr:
+    description: Configure hardware address.
+    required: false
+    type: str
+  monitor_mode:
+    description: set monitor mode for the interface off/on.
+    required: false
+    type: bool
+  mtu:
+    description: interface mtu.
+    required: false
+    type: int
+  name:
+    description: interface name.
+    required: true
+    type: str
+  rx_ringsize:
+    description: Set receive buffer size for interface.
+    required: false
+    type: int
+  speed:
+    description: Interface link speed. Speed is not relevant when 'auto-negotiation' is enabled
+    required: false
+    type: str
+  tx_ringsize:
+    description: Set transmit buffer size for interfaces.
+    required: false
+    type: int
+short_description: Set Physical interface
+version_added: '2.9'
+
+"""
+
+EXAMPLES = """
+- name: Set comment field of a physical interface
+  cp_gaia_physical_interface:
+    comments: eth0 interface
+    enabled: true
+    name: eth0
+
+"""
+
+RETURN = """
+cp_gaia_physical_interface:
+  description: The updated interface details.
+  returned: always.
+  type: dict
+"""
+
+import ast
+import sys
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.checkpoint.checkpoint import api_call_gaia
+
+
+def main():
+    # arguments for the module:
+    fields = ast.literal_eval("""{"monitor_mode": {"required": False, "type": "bool"}, "ipv6_address": {"required": False, "type": "str"},
```

Comment:
Ansible std of defining the `argument_spec` is in followinf manner: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/checkpoint/checkpoint_host.py#L155, plz update this as pointed and remove the imports made for implementing argument_spect currently

Reply:
Thanks, updated.